pkg/kubelet/kuberuntime/kuberuntime_gc_test.go

```diff
@@ -52,55 +68,49 @@ func TestSandboxGC(t *testing.T) {
 		remain      []int               // template indexes of remaining sandboxes
 	}{
 		{
-			description: "sandbox with no containers should be garbage collected.",
+			description: "sandboxes without containers for deleted pods should be garbage collected.",
 			sandboxes: []sandboxTemplate{
-				{pod: pods[0], state: runtimeapi.PodSandboxState_SANDBOX_NOTREADY},
+				makeGCSandbox(pods[2], 0, runtimeapi.PodSandboxState_SANDBOX_NOTREADY),
+				makeGCSandbox(pods[2], 0, runtimeapi.PodSandboxState_SANDBOX_READY),
```

Comment:
Not sure whether we should remove READY sandbox. `SyncLoop` may be stopping the sandbox, too. This may introduce race that sandbox is stopped twice.